### ISSUE TITLE
[WhatsApp] unable to create passkey

### ISSUE THREAD

**Issue by Alicele** [microg/GmsCore#2594]
**Describe the bug**
In the WhatsApp app -> Settings -> Account -> Passkeys -> Create Passkey, I clicked the Use current device screen lock option and verified the fingerprint information as required, but was prompted with an offline pop-up window.

**To Reproduce**
1. Go to  WhatsApp app -> Settings -> Account -> Passkeys -> Create Passkey.
2. Click on Use this device with screen lock.
3. Verified the fingerprint information.
4. Show offline pop-up window.

**Expected behavior**
It will be the same as the pixel phone, showing the content after the key is successfully created.

**Screenshots**
![image](https://github.com/user-attachments/assets/180cadc9-cb9c-4447-bc8a-a18922197466)

**System**
Android Version: [ 12 ]
Custom ROM:  [ HarmonyOS 2.0.0 ]
Device: HUAWEI Mate 60 Pro [ ALN-AL00 ]

**Additional context**

**Comment by Idcrafter** [comment]
i also have this issue on many devices using roms that use CalyxOS.
Pixel 6 Pro(CalyxOS), Pixel 9 Pro XL(CalyxOS), Redmi note 10 (crDroid with flashed microG), Waydroid (arm64, X86_64 also with microG) and a Lumia 950 with lineageOS and microG.

none of them are able to do that in whatsapp.
microG versions are:
CalyxOS (0.3.2.240913-21)
the rest : (0.3.3.240913)

**Comment by 000exploit** [comment]
Something identical with Firefox on LineageOS 20

**Comment by ale5000-git** [comment]
Hve all of you tried with microG Services 0.3.6 and microG Companion 0.3.6?

**Comment by Sid127** [comment]
made a passkey today on whatsapp, 0.3.6

**Comment by Idcrafter** [comment]
i can't make one on CalyxOS with microG 0.3.6.

**Comment by mar-v-in** [comment]
This likely is related to the Android version you use, so please always also report Android version, not only microG version.

**Comment by Idcrafter** [comment]
i use Android 15 CalyxOS 6.2.2 on a Google Pixel 9 Pro XL with a locked bootloader.

**Comment by Idcrafter** [comment]
i will also state some observations after testing passkeys with and without hardware keys. 
everything is tested on Android 15 and microG 0.3.6.244735 i am not logged into any google account in microG:
1. for some reason does google think microG's internal passkey storage is a icloud thing
2. i get prompted for the security key selection activity and can register the fingerprint or hardware key but then will it not work for some apps or services.
3. for whatsapp does it not even give me the prompt inside microG and it just fails like it also does for other apps like the Amazon app and also paypal app and more.
4. microG does not yet seem to offer any kind of key backupping like into a keepass or from device to device via stuff like usb, bt, nfc or wifi (this would be a great addition but could pose a security risk)

but i am impressed  that passkeys already work even for internal keys and am happy to see how it will progress.

**Comment by lucasmz-dev** [comment]
@Idcrafter if you use CalyxOS I suggest going into their rooms for support first
https://calyxos.org/community/

**Comment by Idcrafter** [comment]
hi, this won't work unless microG supports the google password manager infrastructure. whatsapp seems to create a passkey to be stored only in a password manager even on real google play services will it fail when google isn't being used as credential manager.
it probably is required that you can take the passkey with you onto a another device so would the in exportable ones microG and play services make not work here.
i hope this helps.

**Comment by ale5000-git** [comment]
There have been improvements regarding Fido inside microG v0.3.10.
Has anyone tried with microG Services v0.3.10 and microG Companion v0.3.10?

**Comment by Idcrafter** [comment]
yes i tried this on Iode OS 6.9 with microG 0.3.10.250932 on a Pixel 6 Pro (setup as the main whatsapp device) and a Pixel 6a(set up as companion device) and it first prompted me to use keepass to save it and after disabling keepass did i try again but then did it just fail to do anything and notified me that something has gone wrong but it didn't show the popup from the image the OP had posted.

**Comment by lucasmz-dev** [comment]
Do you use a password manager because I think WhatsApp is using the up to date APIs which let password managers like Bitwarden work fine

BTW I have not found like one place where you can actually use this passkey, seems more like a liability ngl

**Comment by Idcrafter** [comment]
Yes i do use Keepass and did get a second number to test this and it just makes the Passkey to authenticate yourself on another device if you migrate the number probably. the passkey seems only to be used as another factor for your number if you register that number again. it is not even asked for if you set up whatsap web or similar. 

But the issue here is probably that microG has no support for the google password manager api's or a implementation of the api's but with a non google backend. You cannot select microG here:

<img width="1344" height="2073" alt="Image" src="https://github.com/user-attachments/assets/8b21ef1b-dfcd-4e98-b80e-e4af492c2a8f" />
this would only be a nice to have if microG implements a password manager.
